src/viam/ur/module/ur_arm.cpp

```diff
@@ -368,132 +1533,8 @@ void URArm::configure_(const std::unique_lock<std::shared_mutex>& lock, const De
-    // A physical/real arm can only be controlled this way if the dashboard is in remote mode.
-    // Conversely, a simulation arm will always report that the arm is in local mode.
-    // Using this, we can determine whether the arm is a simulation arm by calling dashboard->commandIsInRemoteControl at this point.
-    // we will capture this state to use within the worker thread.
-    current_state_->is_sim = !current_state_->dashboard->commandIsInRemoteControl();
-
-    constexpr char k_script_file[] = "/src/control/external_control.urscript";
-
-    // Now the robot is ready to receive a program
-    auto ur_cfg = urcl::UrDriverConfiguration{};
-    ur_cfg.robot_ip = current_state_->host;
-    ur_cfg.script_file = current_state_->appdir + k_script_file;
-    ur_cfg.output_recipe_file = current_state_->appdir + k_output_recipe;
-    ur_cfg.input_recipe_file = current_state_->appdir + k_input_recipe;
-    ur_cfg.handle_program_state = &reportRobotProgramState;
-    ur_cfg.headless_mode = true;
-    ur_cfg.socket_reconnect_attempts = 1;
-
-    if (!current_ports_) {
-        current_ports_ = new_ports();
-    }
-    ur_cfg.reverse_port = current_ports_->reverse_port;
-    ur_cfg.script_sender_port = current_ports_->script_sender_port;
-    ur_cfg.trajectory_port = current_ports_->trajectory_port;
-    ur_cfg.script_command_port = current_ports_->script_command_port;
-    VIAM_SDK_LOG(debug) << "using reverse_port " << ur_cfg.reverse_port;
-    VIAM_SDK_LOG(debug) << "using script_sender_port " << ur_cfg.script_sender_port;
-    VIAM_SDK_LOG(debug) << "using trajectory_port " << ur_cfg.trajectory_port;
-    VIAM_SDK_LOG(debug) << "using script_command_port " << ur_cfg.script_command_port;
-
-    current_state_->driver.reset(new UrDriver(ur_cfg));
-
-    // define callback function to be called by UR client library when trajectory state changes
-    current_state_->driver->registerTrajectoryDoneCallback(std::bind(&URArm::trajectory_done_cb_, this, std::placeholders::_1));
-
-    // Once RTDE communication is started, we have to make sure to read from the interface buffer,
-    // as otherwise we will get pipeline overflows. Therefore, do this directly before starting your
-    // main loop
-    current_state_->driver->startRTDECommunication();
-    int retry_count = 100;
-    while (read_joint_keep_alive_(false) != UrDriverStatus::NORMAL) {
-        if (retry_count <= 0) {
-            throw std::runtime_error("couldn't get joint positions; unable to establish communication with the arm");
-        }
-        retry_count--;
-        std::this_thread::sleep_for(k_noop_delay);
-    }
-
-    // start background thread to continuously send no-ops and keep socket connection alive
-    VIAM_SDK_LOG(info) << "starting background_thread";
-    current_state_->keep_alive_thread = std::thread(&URArm::keep_alive_, this);
+    VIAM_SDK_LOG(info) << "URArm starting up";
+    current_state_ = state_::create(configured_model_type, cfg, ports_);
```

Comment:
All the work from here on that `configure` previously did is now handled by the state machine.